pubspec.yaml

```diff
@@ -0,0 +1,180 @@
+#
+# ------------------------------------------------------------------------------
+# (3) MIGRATING A NON-WORKSPACE PACKAGE TO USING THE WORKSPACE
+# ------------------------------------------------------------------------------
+# Many packages in this repo are still using a pre-workspace style pubspec.yaml,
+# either with manuallu declared `dependency_overrides` (much of ./tools) or by
+# using pub (./web_sdk, ./lib/web_ui). To migrate a package to the workspace:
+#
+# A. Add the `resolution: workspace` field to the pubspec.yaml.
+# B. Update the minimum SDK version to at least `^3.5.0-294.0.dev`.
+# C. Add the package to the `workspace` field in this file.
+# D. Ensure every dependency in the package is added to the `dependencies` field
+#    in this file, following instructions in (4).
+#
+# Once `dart pub get` is run on the workspace, the package will be resolved as
+# part of the workspace, and the `dependency_overrides` in this file will be
+# applied to the package.
+#
+# ------------------------------------------------------------------------------
+# (4) ADDING DEPENDENCIES TO A PACKAGE IN THIS WORKSPACE
+# ------------------------------------------------------------------------------
+# When adding a dependency to a package in the workspace, add the dependency to
+# the `dependencies` field in this file. If the dependency is located within
+# the repository, use the `path` field to point to the package.
+#
+# If the dependency is a third party package, add it to the
+# `dependency_overrides` field in this file. The `any` version constraint is
+# used to indicate that the version of the package is not important, as it is
+# managed by the `DEPS` file.
+
+name: _engine_workspace
+
+# Required for workspace support.
+environment:
+  sdk: ^3.5.0-294.0.dev
+
+# Declare all packages that are part of the workspace.
+workspace:
+  - tools/engine_tool
+
+# Declare all dependencies that are used by one or more packages.
+#
+# A few notes:
+# 1. There is no distinction between "dependencies" and "dev_dependencies";
+#    those notions are for *publishing* packages, not for managing a workspace.
+#    Specific packages in the workspace itself will declare whether they are
+#    dependencies or dev_dependencies, but here it is a union of both.
+#
+# 2. The `any` version constraint is used to indicate that the version of the
+#    package is not important, as it is managed by the `DEPS` file. In other
+#    words, "if the test pass, ship it".
+#
+# 3. The `path` field is used to indicate the location of the package in the
+#    local repository itself. This is to *only* be used for local packages that
+#    are part of the workspace. Third party dependencies are managed by the
+#    `DEPS` file: they get a "any" version constraint and are overridden in the
+#    `dependency_overrides` section below.
+#
+# While not enforced by tooling, try to keep this list in alphabetical order.
```

Comment:
You could enable: https://dart.dev/tools/linter-rules/sort_pub_dependencies

Reply:
Added a TODO for now, thanks!